pubtools/pulplib/_impl/model/unit/erratum.py

```diff
@@ -388,3 +388,9 @@ class ErratumUnit(Unit):
     )
     """IDs of repositories containing the unit, or ``None`` if this information is unavailable.
     """
+
+    unit_id = pulp_attrib(type=str, pulp_field="_id", default=None)
```

Comment:
I just wonder whether it's possible to have unit_id attr only once in `class Unit` . Or is there some limitation of attrs lib that would disallow proper inheritance?

Reply:
Yeah, attrs limitation, exactly.  The limitation is that you can't define some attrs without defaults, then some attrs with defaults, then some more attrs without defaults.  It means the base class can't add anything with a default, because the subclasses will then crash when trying to define non-defaults.

Basically if you were to list out all of the attrs from the entire class hierarchy in the order they're defined, all the ones with default values have to appear grouped at the end.

It's because py2 function signatures work this way (can't define some arguments with defaults, then more arguments without defaults afterward).  The limitation goes away when py2 doesn't have to be supported any more.